### PR TITLE
adding option to flatten pvc clone during drpolicy creation

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
https://issues.redhat.com/browse/RHSTOR-5742

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
image

File: packages/mco/components/create-dr-policy/create-dr-policy.tsx [hunk 1]
```diff
@@ -78,13 +81,17 @@ const createDRPolicy = (
   policyName: string,
   replicationType: REPLICATION_TYPE,
   syncIntervalTime: string,
+  isFlattenImage: boolean,
```
Comment (GowthamShanmugam) [inline packages/mco/components/create-dr-policy/create-dr-policy.tsx:84]: please remove MCO console work from PV encryption PR

**Comment by TimothyAsirJeyasing** [comment]
/retitle  adding option to flatten pvc clone during drpolicy creation

File: packages/mco/components/create-dr-policy/create-dr-policy.tsx (hunk 1 above)
Comment (TimothyAsirJeyasing) [reply packages/mco/components/create-dr-policy/create-dr-policy.tsx:84]: Sorry by mistake i have given a wrong title, I have corrected the title.

File: locales/en/plugin__odf-console.json [hunk 2]
```diff
@@ -35,6 +35,9 @@
   "Note: If your cluster isn't visible on this list, verify its import status and refer to the steps outlined in the ACM documentation.": "Note: If your cluster isn't visible on this list, verify its import status and refer to the steps outlined in the ACM documentation.",
   "Data Foundation {{ version }} or above must be installed on the managed clusters to setup connection for enabling replication/mirroring.": "Data Foundation {{ version }} or above must be installed on the managed clusters to setup connection for enabling replication/mirroring.",
   "Selected clusters": "Selected clusters",
+  "Advanced settings": "Advanced settings",
+  "Flatten PVC clone (for Data Foundation only)": "Flatten PVC clone (for Data Foundation only)",
+  "Uses volume replication class to allow disaster recovery to protect PVC clones regardless the parent/datasource replication configuration.": "Uses volume replication class to allow disaster recovery to protect PVC clones regardless the parent/datasource replication configuration.",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__odf-console.json:40]: is "volume replication class" a CR ?? if so, add text in correct "kind" format...

File: packages/mco/constants/disaster-recovery.ts [hunk 3]
```diff
@@ -21,6 +21,9 @@ export const Actions = {
   DELETE_DR_POLICY: 'Delete DRPolicy',
 };
 
+export const FLATTEN_LABEL = {
```
Comment (SanjalKatiyar) [inline packages/mco/constants/disaster-recovery.ts:24]: better to add a line or two comment on why is it needed ? if possible add reference link to csi doc...

**Comment by SanjalKatiyar** [comment]
check why tests are failing...

File: packages/mco/components/create-dr-policy/reducer.ts [hunk 4]
```diff
@@ -99,6 +103,12 @@ export const drPolicyReducer = (
         selectedClusters: action.payload,
       };
     }
+    case DRPolicyActionType.SET_FLAT_IMAGE: {
```
Comment (GowthamShanmugam) [inline packages/mco/components/create-dr-policy/reducer.ts:106]:
```suggestion
    case DRPolicyActionType.ENABLE_RBD_IMAGE_FLATTEN: {
```

File: packages/mco/constants/disaster-recovery.ts (hunk 3 above)
Comment (GowthamShanmugam) [inline packages/mco/constants/disaster-recovery.ts:24]:
```suggestion
export const RBD_IMAGE_FLATTEN_LABEL = {
```

File: locales/en/plugin__odf-console.json (hunk 2 above)
Comment (TimothyAsirJeyasing) [reply locales/en/plugin__odf-console.json:40]: sure i will update the text with VolumeReplicationClass
I can see OpenShift docs, they have used VolumeReplicationClass. 
LINK - https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.9/html/configuring_openshift_data_foundation_for_regional-dr_with_advanced_cluster_management/creating-volumereplicationclass-resource_rhodf#:~:text=The%20VolumeReplicationClass%20is%20used%20to,cluster%20to%20the%20remote%20cluster.

File: packages/mco/components/create-dr-policy/create-dr-policy.tsx [hunk 5]
```diff
@@ -119,6 +127,37 @@ const createMirrorPeer = (
 const getDRPolicyListPageLink = (url: string) =>
   url.replace(`/${referenceForModel(DRPolicyModel)}/~new`, '');
 
+const AdvancedSettings: React.FC<{
+  enableRBDImageFlatten: boolean;
+  dispatch: React.Dispatch<DRPolicyAction>;
+}> = ({ enableRBDImageFlatten, dispatch }) => {
```
Comment (bipuladh) [inline packages/mco/components/create-dr-policy/create-dr-policy.tsx:133]: Use types

File: packages/mco/components/create-dr-policy/create-dr-policy.tsx [hunk 6]
```diff
@@ -119,6 +127,37 @@ const createMirrorPeer = (
...
+  const { t } = useCustomTranslation();
+
+  const handleRbdImageFlattenOnChange = (checked: boolean) => {
```
Comment (SanjalKatiyar) [inline packages/mco/components/create-dr-policy/create-dr-policy.tsx:136]:
```suggestion
  const handleRBDImageFlattenOnChange = (checked: boolean) => {
```

File: packages/mco/constants/disaster-recovery.ts [hunk 7]
```diff
@@ -21,6 +21,11 @@ export const Actions = {
   DELETE_DR_POLICY: 'Delete DRPolicy',
 };
 
+// This used to specify a custom VolumeReplicationClass (VRC) for disaster recovery protection of cloned images.
+// Allow CNV users to select a non-default VRC during DRPolicy creation.
```
Comment (SanjalKatiyar) [inline packages/mco/constants/disaster-recovery.ts:25]: we are neither specifying nor allowing user to select any VRC from the UI... it's just a checkbox...
cc @GowthamShanmugam do we need to rephrase this or is it correct ??

File: packages/mco/components/create-dr-policy/create-dr-policy.tsx [hunk 8]
```diff
@@ -78,13 +82,17 @@ const createDRPolicy = (
...
   syncIntervalTime: string,
+  enableRBDImageFlatten: boolean,
   peerNames: string[]
 ) => {
   const drPolicyPayload: DRPolicyKind = {
     apiVersion: getAPIVersionForModel(DRPolicyModel),
     kind: DRPolicyModel.kind,
     metadata: { name: policyName },
     spec: {
+      VolumeReplicationClassSelector: enableRBDImageFlatten
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply packages/mco/components/create-dr-policy/create-dr-policy.tsx:93]: https://github.com/red-hat-storage/ramen/blob/main/api/v1alpha1/drpolicy_types.go

**Comment by alfonsomthd** [comment]
LGTM. Waiting for @SanjalKatiyar for tagging.

**Comment by SanjalKatiyar** [comment]
plz update commit title... it's a diff epic...

**Comment by TimothyAsirJeyasing** [comment]
image

**Comment by SanjalKatiyar** [comment]
> image

better to align checkbox and alert with toggle a bit...

**Pull request merged**